omero_search_engine/validation/results_validator.py

```diff
@@ -581,16 +581,24 @@ def test_no_images():
         "\n\n\n======================== Test number of images inside each study ============================\n"  # noqa
     ]
     for name, numbers in names.items():
-        and_filters = [
-            {
-                "name": "Name (IDR number)",
-                "value": name,
-                "operator": "equals",
-                "resource": "project",
-            }
+        or_filters = [
+            [
+                {
+                    "name": "name",
+                    "value": name,
+                    "operator": "equals",
+                    "resource": "screen",
+                },
+                {
+                    "name": "name",
+                    "value": name,
+                    "operator": "equals",
+                    "resource": "project",
+                },
+            ]
         ]
-        or_filtes = []
-        query = {"and_filters": and_filters, "or_filters": or_filtes}
+        and_filters = []
```

Comment:
This is the major change in this PR i.e. the swap of filters.
Could you expand on the rationale of that change?

Reply:
thanks. I remember the discussion now

Reply:
It is possible to test this PR using the following two examples,

The first one will return the container which meets the search criteria along with the container attribute as it is used `submitquery/containers/` endpoint.

`curl -X POST "https://idr-testing.openmicroscopy.org/searchengine//api/v1/resources/submitquery/containers/" -H "accept: application/json" -H "Content-Type: application/json" -d "{ \"query_details\": { \"and_filters\": [ { \"name\": \"name\", \"value\": \"idr0123-mota-mifish/experimentA\", \"operator\": \"equals\", \"resource\": \"project\" } ], \"or_filters\": [], \"case_sensitive\": false }}"`

The following example uses `submitquery` so it returns the images inside the container which meet the search criteria

`curl -X POST "https://idr-testing.openmicroscopy.org/searchengine//api/v1/resources/submitquery/" -H "accept: application/json" -H "Content-Type: application/json" -d "{ \"query_details\": { \"and_filters\": [ { \"name\": \"name\", \"value\": \"idr0123-mota-mifish/experimentA\", \"operator\": \"equals\", \"resource\": \"project\" } ], \"or_filters\": [], \"case_sensitive\": false }}"`